File: src/frontend/src/app/interviews/models/interview.ts
import { Recruiter } from "../../recruiters/models/recruiter";
import { Vacancy } from "../../vacancies/models/vacancy";
import { InterviewType } from "./interview-type";
import { Candidate } from "../../candidates/models/candidate";

export interface Interview {
  id: number;
  title: string;
  startTime: string;
  endTime: string;
  vacancy: Vacancy;
  recruiter: Recruiter;
  candidate: Candidate;
  interviewType: InterviewType;
}

File: src/api/BoardingTracker.Domain/Entities/Interview.cs
namespace BoardingTracker.Domain.Entities
{
    public partial class Interview
    {
        public int Id { get; set; }
        public string Title { get; set; } = null!;
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public int? VacancyId { get; set; }
        public Guid? RecruiterId { get; set; }
        public Guid? CandidateId { get; set; }
        public int? InterviewTypeId { get; set; }

        public virtual Candidate? Candidate { get; set; }
        public virtual InterviewType? InterviewType { get; set; }
        public virtual Recruiter? Recruiter { get; set; }
        public virtual Vacancy? Vacancy { get; set; }
    }
}
